src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,147 @@
+        }
+        array = temporaryArray;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int indexOfArrayList = 0;
+        if ((sizeOfArray + list.size()) >= array.length) {
+            temporaryArray = grow(sizeOfArray + list.size());
+        } else {
+            temporaryArray = (T[])(new Object[sizeOfArray + list.size()]);
+        }
+        for (int i = 0; i < sizeOfArray; i++) {
+            temporaryArray[i] = array[i];
+        }
+        for (int i = sizeOfArray; i < sizeOfArray + list.size(); i++) {
+            temporaryArray[i] = list.get(indexOfArrayList);
+            indexOfArrayList++;
+        }
+        array = temporaryArray;
+        sizeOfArray += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= sizeOfArray || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_OUT_OF_BOUNDS_MASSAGE);
+        }
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= sizeOfArray || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_OUT_OF_BOUNDS_MASSAGE);
+        }
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= sizeOfArray || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(INDEX_OUT_OF_BOUNDS_MASSAGE);
+        }
+        temporaryArray = (T[])(new Object[array.length]);
+        for (int i = 0; i < index; i++) {
+            temporaryArray[i] = array[i];
+        }
+        for (int i = index + 1; i < sizeOfArray; i++) {
+            temporaryArray[i - 1] = array[i];
+        }
+        T removedValue = array[index];
+        array = temporaryArray;
+        sizeOfArray--;
+        return removedValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int indexOfElement = 0;
+        temporaryArray = (T[])(new Object[array.length]);
+        for (int i = 0; i < sizeOfArray; i++) {
+            if (Objects.equals(element, array[i]) || indexOfElement == sizeOfArray) {
+                break;
+            }
+            indexOfElement++;
+        }
+        if (indexOfElement == sizeOfArray) {
+            throw new NoSuchElementException(INDEX_OUT_OF_BOUNDS_MASSAGE);
+        }
+        for (int i = 0; i < indexOfElement; i++) {
+            temporaryArray[i] = array[i];
+        }
+        for (int i = indexOfElement + 1; i < sizeOfArray; i++) {
+            temporaryArray[i - 1] = array[i];
+        }
+        array = temporaryArray;
+        sizeOfArray--;
+        return element;
     }
 
     @Override
     public int size() {
-        return 0;
+        return sizeOfArray;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return sizeOfArray == 0;
+
+    }
+
+    private T[] grow(int nextSizeOfArray) {
```

Comment:
```suggestion
    private void grow() {
```

Reply:
Simplify your method, you don`t need to return any from this method

Reply:
Done.